Blizzard.IO.RabbitMQ/NetqRabbitConsumer.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using Blizzard.IO.Core;
+using Blizzard.IO.RabbitMQ.Entities;
+using EasyNetQ;
+using EasyNetQ.Topology;
+using Microsoft.Extensions.Logging;
+
+namespace Blizzard.IO.RabbitMQ
+{
+    public class NetqRabbitConsumer<TData> : IConsumer<TData>, IConsumerWithMetadata<TData, RabbitMessageProperties>
+    {
+        private readonly IBus _netqBus;
+        private readonly IDeserializer<TData> _deserializer;
+        private readonly IAbstractTypeDeserializer<TData> _concreteTypeDeserializer;
+        private readonly IQueue _netqQueue;
+        private readonly IConverter<MessageProperties, RabbitMessageProperties> _converter;
+        private readonly ILogger<NetqRabbitConsumer<TData>> _logger;
+        
+        private IDisposable _consumeHandler;
+        private readonly Func<byte[], MessageProperties, TData> _dataExtractionStrategy;
+
+        public event Action<TData> MessageReceived;
+        public event Action<TData, RabbitMessageProperties> MessageWithMetadataReceived;
+
+        protected NetqRabbitConsumer(IBus netqBus, RabbitQueue sourceQueue, ILoggerFactory loggerFactory,
+            IConverter<MessageProperties, RabbitMessageProperties> coverter)
+        {
+            _netqBus = netqBus;
+            _netqQueue = new Queue(sourceQueue.Name, sourceQueue.Exclusive);
+            _logger = loggerFactory.CreateLogger<NetqRabbitConsumer<TData>>();
+            _converter = coverter ?? new RabbitPropertiesConverter();
+        }
+
+        public NetqRabbitConsumer(IBus netqBus, RabbitQueue sourceQueue, IDeserializer<TData> deserializer, 
+            ILoggerFactory loggerFactory, IConverter<MessageProperties, RabbitMessageProperties> coverter = null)
```

Comment:
Again if we are allowing to inject converter we need to allow it on both sides (publisher and conusmer).
so talk about it with yair @ybs007

Reply:
yair allowed it on his side, as far as I can see